[quant] Skip copy_same_type_transpose_ for quantized tensor

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29660 [quant] Guard against copying from quantized Tensor to non-quantized Tensor
* **#29609 [quant] Skip copy_same_type_transpose_ for quantized tensor**

Summary:
We can enable this path later if there is a need.
trying to fix: https://github.com/pytorch/pytorch/issues/29435

Test Plan:
python test/test_quantized_tensor.py
Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18453723](https://our.internmc.facebook.com/intern/diff/D18453723)

jerryzh168 committed:
[quant] Skip copy_same_type_transpose_ for quantized tensor

Summary:
We can enable this path later if there is a need.

Test Plan:
python test/test_quantized_tensor.py
Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on test/test_quantized_tensor.py:
```diff
@@ -181,6 +181,11 @@ def test_qtensor_permute(self):
+        # permuting larger tensors
+        x = torch.randn(64, 64)
+        qx = torch.quantize_per_tensor(x, 1.0, 0, torch.qint32)
+        print(qx.permute([1, 0]))
```
I think it's better to use the `dequantize` instead of `print`:

```python
# permuting larger tensors
x = torch.randn(64, 64)
qx = torch.quantize_per_tensor(x, 1.0, 0, torch.qint32)
qx.permute([1, 0]).dequantize()
```

jerryzh168 replied:
oh..we can just call this function here

z-a-f reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[quant] Skip copy_same_type_transpose_ for quantized tensor"


Summary:
We can enable this path later if there is a need.
trying to fix: https://github.com/pytorch/pytorch/issues/29435

Test Plan:
python test/test_quantized_tensor.py
Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
